tests/tools/s3_download.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/python3
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+"""Script for downloading the resources required for tests.
+It returns the absolute path of the downloaded resource to stdout."""
+
+import json
+from optparse import OptionParser
+import os
+import pathlib
+
+from utils import S3ResourceFetcher
+
+if __name__ == "__main__":
+    parser = OptionParser()
+    parser.add_option("-t", "--resource-type", dest="resource_type",
+                      help="Type of resource to download.")
+    parser.add_option("-n", "--resource-name",
+                      dest="resource_name",
+                      help="Name of resource to download.")
+    # This hack to have both --tags and -o is needed because running
+    # a command in Rust with std::process::Command does not work with
+    # parameters that have double dashes.
+    parser.add_option("--tags",
+                      dest="tags",
+                      help="The optional tags that the resource must have as a dictionary.",
+                      default="{}")
+    parser.add_option("-p", "--path",
+                      dest="path",
+                      help="The directory where to save the downloaded resource. "
+                           "This path represents the root used when creating the "
+                           "structure. The full path is created by joining this path "
+                           "and the relative_path defined in the resource manifest.")
+    (options, args) = parser.parse_args()
+
+    required_ops = ["resource_type", "resource_name"]
```

Comment:
Better to check for `options.resource_type is None and options.resource_name is None`. More Pythonic.

Reply:
If we add more required fields than this check will become rather large. I prefer to have a `required_ops` list.

Reply:
Yes, makes sense. Can we do `getattr(options, op)` instead of `options.__dict__` . Somehow it's best to avoid using the internal `__dict__`  if we can. But that's more like a nit.